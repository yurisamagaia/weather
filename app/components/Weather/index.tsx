import React, { FC } from 'react';
import { Image } from 'react-native'
import { getWeatherKinds, convertMetersToKm } from '../../utils/kinds';
import { WeatherData } from '../../store/actions/types'
import {
  Container,
  Content,
  IconContainer,
  TextCity,
  TextTemp,
  TextDescription,
  Title,
  ContentFlex,
  Temperature,
  TextCondition
} from './style';

type Props = {
  data: | WeatherData
};

const Weather: FC<Props> = ({ data }) => (
  <Container>
    <Content>
      <TextCity>{data.city} - {data.country}</TextCity>
      <ContentFlex>
        <Content>
          <Temperature>{Math.round(data.temperature)}째</Temperature>
          <ContentFlex>
            <TextTemp>H: {Math.round(data.temp_max)}째</TextTemp>
            <TextTemp>L: {Math.round(data.temp_min)}째</TextTemp>
          </ContentFlex>
        </Content>
        <IconContainer>
          <Image
            source={{ uri: 'https://openweathermap.org/img/wn/' + data.icon + '@4x.png' }}
            style={{ width: 200, height: 200 }}
          />
        </IconContainer>
      </ContentFlex>
    </Content>
    <Content>
      <Title>{data.title}</Title>
      <TextCondition>Feels like: {Math.round(data.feels_like)}째</TextCondition>
      <TextCondition>Wind: {convertMetersToKm(data.wind)}km/h</TextCondition>
      <TextCondition>Humidity: {data.humidity}%</TextCondition>
      <TextCondition>Pressure: {data.pressure}hPa</TextCondition>
    </Content>
    <TextDescription>{getWeatherKinds(data.kind)}</TextDescription>
  </Container>
);

export default Weather;