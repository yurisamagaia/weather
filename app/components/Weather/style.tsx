// import styled from 'styled-components/native';
// import { colors } from '../../theme';

// export const Container = styled.View`
//   flex: 1;
// `;

// export const Content = styled.View`
//   flex: 1;
//   align-items: center;
// `;

// export const IconContainer = styled.View`
//   align-self: center;
// `;

// export const TextCity = styled.Text`
//   font-size: 22px;
//   color: ${colors.white};
// `;

// export const TextTemp = styled.Text`
//   font-size: 50px;
//   color: ${colors.white};
// `;

// export const TextDescription = styled.Text`
//   font-size: 20px;
//   text-align: center;
//   color: ${colors.white};
//   font-style: italic;
//   margin-top: 10px;
// `;

// export const Title = styled.Text`
//   font-size: 22px;
//   color: ${colors.white};
//   padding-bottom: 10px;
// `;

import styled from 'styled-components/native';
import { colors } from '../../theme';

export const Container = styled.View`
  flex: 1;
`;

export const Content = styled.View`
  
`;

export const ContentFlex = styled.View`
  width: 100%;
  align-items: center;
  flex-direction: row;
  
`;

export const IconContainer = styled.View`
  align-self: center;
`;

export const TextCity = styled.Text`
  font-size: 22px;
  color: ${colors.white};
`;

export const Temperature = styled.Text`
  font-size: 60px;
  font-weight: 200;
  color: ${colors.white};
`;

export const TextTemp = styled.Text`
  font-size: 16px;
  font-weight: 400;
  color: ${colors.white};
  margin-right: 6px;
`;

export const ContainerTextCondition = styled.View`
  align-items: flex-start;
`;
  
export const TextCondition = styled.Text`
  font-size: 18px;
  font-weight: 400;
  color: ${colors.white};
  margin-top: 5px;
`;

export const TextDescription = styled.Text`
  font-size: 16px;
  text-align: center;
  color: ${colors.white};
  font-style: italic;
  margin-top: 40px;
`;

export const Title = styled.Text`
  font-size: 26px;
  color: ${colors.white};
  padding-bottom: 10px;
`;