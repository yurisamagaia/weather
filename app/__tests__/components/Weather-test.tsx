import React from 'react';
import renderer from 'react-test-renderer';
import Weather from '../../components/Weather';

test('Should render correctly', async () => {
  const weather = {
    kind: 'few clouds',
    title: 'Clouds',
    icon: '03d',
    temperature: 28,
    city: 'Angra dos Reis',
    country: 'BR',
  };
  const component = renderer.create(<Weather data={weather} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});