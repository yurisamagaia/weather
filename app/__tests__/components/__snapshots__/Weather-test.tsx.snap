// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 22,
          },
        ]
      }
    >
      Angra dos Reis
       - 
      BR
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontSize": 60,
                "fontWeight": "200",
              },
            ]
          }
        >
          28
          째
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "marginRight": 6,
                },
              ]
            }
          >
            H: 
            30
            째
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "marginRight": 6,
                },
              ]
            }
          >
            L: 
            24
            째
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://openweathermap.org/img/wn/03d@4x.png",
            }
          }
          style={
            Object {
              "height": 200,
              "width": 200,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 26,
            "paddingBottom": 10,
          },
        ]
      }
    >
      Clouds
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 5,
          },
        ]
      }
    >
      Feels like: 
      29
      째
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 5,
          },
        ]
      }
    >
      Wind: 
      14
      km/h
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 5,
          },
        ]
      }
    >
      Humidity: 
      72
      %
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 5,
          },
        ]
      }
    >
      Pressure: 
      1003
      hPa
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 16,
          "fontStyle": "italic",
          "marginTop": 40,
          "textAlign": "center",
        },
      ]
    }
  >
    It's a good day to be outside, but be careful, there are some clouds.
  </Text>
</View>
`;
